src/addons/consoleCatcher.ts

```diff
@@ -13,6 +13,73 @@ const createConsoleCatcher = (): {
   const consoleOutput: ConsoleLogEvent[] = [];
   let isInitialized = false;
 
+  /**
+   * Safely stringifies objects handling circular references
+   */
+  const safeStringify = (obj: unknown): string => {
```

Comment:
I'd suggest using `safe-stringify` since it is almost the same, but covered by tests